### PR TITLE
Baseurl

### PR DESCRIPTION
**Pull request by lookfirst**
This enabled me to put the system.config.js file into the src folder and fixed an issue where the promises were not being returned correctly.

### CODE REVIEW

**Comment by lookfirst** [comment]
Hmm... I'm noticing that one of my commits isn't really helping anything. 5220af3

What I'm seeing is that running minify after using this code is still broken. There is some promise that isn't finishing before the end again for some reason.

File: lib/builder.js [hunk 1]
```diff
@@ -178,17 +179,24 @@ var buildTrees = function (bundles, config) {
...
+  return builder.buildTree(tree.tree, baseURL + '/' + destination + '.js', {
     sourceMaps: config.sourceMaps,
     minify: config.minify,
     mangle: config.mangle,
-    config: config.config
+    config: config.config,
+    main: config.main,
+    dest: config.dest,
+    destJs: config.destJs,
+    baseURL: config.baseURL
```
Comment (marjan-georgiev) [inline lib/builder.js]: Why are the last 4 lines here passed to buildTree?

File: lib/builder.js [hunk 2]
```diff
@@ -178,17 +179,24 @@ var buildTrees = function (bundles, config) {
 }
 
 var buildTree = function (tree, destination, config) {
-  if (destination.indexOf('systemjs/plugin') != -1) {
+  if (destination.indexOf('systemjs/plugin') != -1 ||
+    destination.indexOf('bower_components') != -1) {
```
Comment (marjan-georgiev) [inline lib/builder.js]: bower_components should not be ignored here. We want to bundle them together with the app code.

**Comment by marjan-georgiev** [comment]
This commit actually breaks the build: https://github.com/lookfirst/systemjs-route-bundler/commit/5220af3551449b7d2fd985c413eba279304a6a82

It should work without it. See current master.

**Comment by lookfirst** [comment]
Ok... it should be ready to go again. I was up late and wasn't thinking straight.

**Comment by lookfirst** [comment]
don't merge this yet

**Comment by lookfirst** [comment]
ok, please merge this now. i've successfully got my seed project working with everything under dist now, so this is definitely working for me and shouldn't impact your code at all.

**Comment by marjan-georgiev** [comment]
:+1:

**Pull request merged**